app/services/container_dashboard_service.rb

```diff
@@ -175,8 +173,54 @@ def ems_utilization
           :yData => ["used"] + used_mem.values.map { |m| (m / 1024.0).round }
         }
       }
-    else
-      {}
     end
   end
+
+  def hourly_network_trends
+    resource_ids = @ems.present? ? [@ems.id] : ManageIQ::Providers::ContainerManager.all.pluck(:id)
+    hourly_network_trend = Hash.new(0)
+    hourly_metrics =
+      MetricRollup.find_all_by_interval_and_time_range("hourly", 24.hours.ago.beginning_of_hour.utc, Time.now.utc)
+    hourly_metrics =
+      hourly_metrics.where('resource_type = ? AND resource_id in (?)', 'ExtManagementSystem', resource_ids)
```

Comment:
@zeari 

> If you note, that becomes a sub query. Not sure if you want to bring these ids back, but I'm guessing not.

A `pluck(:id)` brings back all the ids and then ships them back to the server. The code I included does not bring back all the ids but rather produces a sub query. Something like that would be preferred.

> Also, is there a way to limit the resource type rather than joining to the ems table?

I noticed in a few places (not this code) we are joining to the ems table to limit the type of records brought back. But the table itself had an STI type column and we could have just limited based upon that value in the first place. Here, I don't think it would work. And joining to the ems table makes more sense because it is small, rather than to a bigger table.